app/src/main/java/org/dieschnittstelle/mobile/android/skeleton/viewmodel/MainViewModel.java

```diff
@@ -0,0 +1,90 @@
+package org.dieschnittstelle.mobile.android.skeleton.viewmodel;
+
+import android.util.Log;
+import android.view.inputmethod.EditorInfo;
+
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+
+import org.dieschnittstelle.mobile.android.skeleton.model.IUserDatabaseOperation;
+import org.dieschnittstelle.mobile.android.skeleton.model.User;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MainViewModel extends ViewModel {
+    private IUserDatabaseOperation userDBOperation;
+    private final MutableLiveData<LoginState> loginState = new MutableLiveData<>();
+    private MutableLiveData<String> mailInputError = new MutableLiveData<>();
+    private MutableLiveData<String> passwordInputError = new MutableLiveData<>();
+    public User user;
+
+    public void authenticateUser() {
+        if ((checkMailInput() == true) && (checkPasswordInput() == true)) {
+            Log.i("Login", "auth");
+            new Thread(() -> {
+                if (userDBOperation.authenticateUser(user.getEmail(), user.getPwd())) {
+                    loginState.postValue(LoginState.AUTHENTICATION_SUCCESS);
+                } else {
+                    loginState.postValue(LoginState.AUTHENTICATION_FAIL);
+                }
+            }).start();
+        }
+    }
+
+    public boolean checkMailInput() {
+        Log.i("Login", "checkMailInputOnEnterKey");
+        if (!(android.util.Patterns.EMAIL_ADDRESS.matcher(user.getEmail()).matches())) {
+            this.mailInputError.setValue("Input not valid, must be an e-mail address");
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    public boolean checkPasswordInput() {
+        boolean pwdCheck = Pattern.matches("[0-9]{6}", user.getPwd());
+        if (!(pwdCheck)) {
```

Comment:
es fehlt davor noch die Überprüfung ob der Benutzer Enter oder tab drückt (siehe TaskViewModel)

```suggestion
        if (keyId == EditorInfo.IME_ACTION_NEXT || keyId == EditorInfo.IME_ACTION_DONE) {
            if (!(pwdCheck)) {...}
        }

```